fix case clause error in verify! due to pop_expect setting invalid state

ppraisethesun opened the pull request:
Error was raised when using stub together with expect 
```     
     ** (FunctionClauseError) no function clause matching in anonymous fn/1 in Promox.verify!/1

     The following arguments were given to anonymous fn/1 in Promox.verify!/1:
     
         # 1
         {{Protocol, :callback, 1}, nil}
```

ppraisethesun committed:
fix case clause error in verify! due to pop_expect setting invalid state

leggebroten commented:
I'm having this exact issue, this repo being maintained any longer?

dsdshcym commented:
I'm sooooooooo sorry that I missed this PR
@ppraisethesun It must be frustrating to see this PR hanging around for so long

@leggebroten and I can't imagine the bad feeling if this issue blocks your tests from passing

I'm merging this PR and publishing a new release ~right now~ today
Again, apologize for the long delay

dsdshcym committed:
Revert "fix case clause error in verify! due to pop_expect setting invalid state"

This reverts commit ae0f440f2c5e59577682b1d1ec7589c3a8f99bab.

This change would break the expected behavior:
- get_and_update_in should not change the value `nil` to `{[], []}`,
  because `nil` means that this pfa is not `expect`ed,
  but `{[], []}` means that this pfa is `expect`ed to be called 0 times.

dsdshcym committed:
Check UnexpectedCallError again in Promox.verify!

- in case the UnexpectedCallError was rescued before,
  this is the only case that I can think of that caused
  https://github.com/dsdshcym/promox/pull/1

dsdshcym committed:
Add tests for a function that's expected to be called 0 times

- cover cases if we don't separate `nil` from `{[], []}` in the state
  (which is the reason behind [1: 9b1259f] )

1: 2022-04-05 9b1259fdb91ef9d9c1b2e85f20bd37484ff3fa50
   Revert "fix case clause error in verify! due to pop_expect setting invalid state"

dsdshcym commented:
Hi @ppraisethesun @leggebroten 
I've published https://hex.pm/packages/promox/0.1.3 which should fix this issue
Please let me know if you still see this error and I promise I won't miss the notification this time
Sorry again for the late response

dsdshcym commented:
Hi @ppraisethesun @leggebroten 
I've published https://hex.pm/packages/promox/0.1.3 which should fix this issue
Please let me know if you still see this error and I promise I won't miss the notification this time
Sorry again for the late response
